waku/v2/protocol/waku_swap/waku_swap.nim

```diff
@@ -146,6 +147,14 @@ proc handleCheque*(ws: WakuSwap, cheque: Cheque) =
   let peerOpt = ws.peerManager.selectPeer(WakuSwapCodec)
   let peerId = peerOpt.get().peerId
 
+  # Get the original signer using web3. For now, a static value (0x6C3d502f1a97d4470b881015b83D9Dd1062172e1) will be used.
+  # Check if web3.eth.personal.ecRecover(messageHash, signature); or an equivalent function has been implemented in nim-web3
```

Comment:
You need something like https://github.com/vacp2p/swap-contracts-module/blob/master/src/swap-helpers.js#L132 don't think it is nim-web3

Reply:
Nice. I'll need to create a PR in the swap Contracts module for adding a `verifySignature` function similar to what was done [here](https://docs.ethers.io/v4/cookbook-signing.html#signing-a-string-message). Essence of this function is to return the public key of the signer. 
For now, I'll assume the returned address is `0x6C3d502f1a97d4470b881015b83D9Dd1062172e1`